From 47f19470c109323c9f9dcf63b873bf1aa7dfb24c Mon Sep 17 00:00:00 2001
From: zigvm maintainers <maintainers@zigvm.local>
Date: Mon, 3 Aug 2026 13:15:00 +0200
Subject: [PATCH] compat: migrate ppx_deriving 6.1.3 to the OCaml 5.5 AST

OCaml 5.5 represents alias names as strings and Pexp_function as a case
list. ppxlib 0.38 exposes those shapes directly.
---
 src/api/ppx_deriving.cppo.ml             | 2 +-
 src_plugins/eq/ppx_deriving_eq.ml        | 2 +-
 src_plugins/fold/ppx_deriving_fold.ml    | 2 +-
 src_plugins/iter/ppx_deriving_iter.ml    | 2 +-
 src_plugins/map/ppx_deriving_map.ml      | 2 +-
 src_plugins/ord/ppx_deriving_ord.ml      | 2 +-
 6 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/api/ppx_deriving.cppo.ml b/src/api/ppx_deriving.cppo.ml
index e0d982a..a4e3966 100644
--- a/src/api/ppx_deriving.cppo.ml
+++ b/src/api/ppx_deriving.cppo.ml
@@ -465,7 +465,7 @@ let free_vars_in_core_type typ =
     | { ptyp_desc = (Ptyp_tuple xs | Ptyp_constr (_, xs)) } ->
       List.map free_in xs |> List.concat
     | { ptyp_desc = Ptyp_alias (x, name) } ->
-      [mkloc name.txt typ.ptyp_loc]
+      [mkloc name typ.ptyp_loc]
       @ free_in x
     | { ptyp_desc = Ptyp_poly (bound, x) } ->
       List.filter (fun y -> not (List.mem y bound)) (free_in x)
diff --git a/src_plugins/eq/ppx_deriving_eq.ml b/src_plugins/eq/ppx_deriving_eq.ml
index cc35ac5..9902741 100644
--- a/src_plugins/eq/ppx_deriving_eq.ml
+++ b/src_plugins/eq/ppx_deriving_eq.ml
@@ -183,7 +183,7 @@ let str_of_type ({ ptype_loc = loc } as type_decl) =
     (* Ensure expr is statically constructive by eta-expanding non-funs.
        See https://github.com/ocaml-ppx/ppx_deriving/pull/252. *)
     match expr with
-    | { pexp_desc = Pexp_function (_ :: _, _, _); _ } -> expr
+    | { pexp_desc = Pexp_function (_ :: _); _ } -> expr
     | _ -> [%expr fun x -> [%e expr] x]
   in
   let out_type =
diff --git a/src_plugins/fold/ppx_deriving_fold.ml b/src_plugins/fold/ppx_deriving_fold.ml
index a592bf0..27cbd68 100644
--- a/src_plugins/fold/ppx_deriving_fold.ml
+++ b/src_plugins/fold/ppx_deriving_fold.ml
@@ -74,7 +74,7 @@ let rec expr_of_typ typ =
     [%expr fun acc -> [%e Exp.function_ cases]]
   | { ptyp_desc = Ptyp_var name } -> evar ("poly_"^name)
   | { ptyp_desc = Ptyp_alias (typ, name) } ->
-    [%expr fun acc x -> [%e evar ("poly_"^name.txt)] ([%e expr_of_typ typ] acc x) x]
+    [%expr fun acc x -> [%e evar ("poly_"^name)] ([%e expr_of_typ typ] acc x) x]
   | { ptyp_loc } ->
     raise_errorf ~loc:ptyp_loc "%s cannot be derived for %s"
                  deriver (Ppx_deriving.string_of_core_type typ)
diff --git a/src_plugins/iter/ppx_deriving_iter.ml b/src_plugins/iter/ppx_deriving_iter.ml
index c0798ea..4b31388 100644
--- a/src_plugins/iter/ppx_deriving_iter.ml
+++ b/src_plugins/iter/ppx_deriving_iter.ml
@@ -70,7 +70,7 @@ let rec expr_of_typ typ =
     Exp.function_ cases
   | { ptyp_desc = Ptyp_var name } -> [%expr ([%e evar ("poly_"^name)] : [%t Typ.var name] -> unit)]
   | { ptyp_desc = Ptyp_alias (typ, name) } ->
-    [%expr fun x -> [%e evar ("poly_"^name.txt)] x; [%e expr_of_typ typ] x]
+    [%expr fun x -> [%e evar ("poly_"^name)] x; [%e expr_of_typ typ] x]
   | { ptyp_loc } ->
     raise_errorf ~loc:ptyp_loc "%s cannot be derived for %s"
                  deriver (Ppx_deriving.string_of_core_type typ)
diff --git a/src_plugins/map/ppx_deriving_map.ml b/src_plugins/map/ppx_deriving_map.ml
index 6148e05..24c4abf 100644
--- a/src_plugins/map/ppx_deriving_map.ml
+++ b/src_plugins/map/ppx_deriving_map.ml
@@ -76,7 +76,7 @@ let rec expr_of_typ ?decl typ =
     Exp.function_ cases
   | { ptyp_desc = Ptyp_var name } -> evar ("poly_"^name)
   | { ptyp_desc = Ptyp_alias (typ, name) } ->
-    [%expr fun x -> [%e evar ("poly_"^name.txt)] ([%e expr_of_typ ?decl typ] x)]
+    [%expr fun x -> [%e evar ("poly_"^name)] ([%e expr_of_typ ?decl typ] x)]
   | { ptyp_loc } ->
     raise_errorf ~loc:ptyp_loc "%s cannot be derived for %s"
                  deriver (Ppx_deriving.string_of_core_type typ)
diff --git a/src_plugins/ord/ppx_deriving_ord.ml b/src_plugins/ord/ppx_deriving_ord.ml
index 132b3b5..94243d5 100644
--- a/src_plugins/ord/ppx_deriving_ord.ml
+++ b/src_plugins/ord/ppx_deriving_ord.ml
@@ -219,7 +219,7 @@ let str_of_type ({ ptype_loc = loc } as type_decl) =
     (* Ensure expr is statically constructive by eta-expanding non-funs.
        See https://github.com/ocaml-ppx/ppx_deriving/pull/252. *)
     match expr with
-    | { pexp_desc = Pexp_function (_ :: _, _, _); _ } -> expr
+    | { pexp_desc = Pexp_function (_ :: _); _ } -> expr
     | _ -> [%expr fun x -> [%e expr] x]
   in
   let out_type =
-- 
2.51.2
